apps/omg_child_chain/test/omg_child_chain/api/transaction_test.exs

```diff
@@ -0,0 +1,55 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.ChildChain.API.TransactionTest do
+  use ExUnit.Case, async: true
+  alias OMG.ChildChain.API.Transaction
+
+  @child_chain __MODULE__.MockChildChain
+  @telemetry __MODULE__.MockTelemetry
+
+  describe "submit/1" do
+    test "emits a :submit and :submit_success telemetry event on a successful submission" do
+      {:ok, _} = Transaction.submit(<<1, 1, 1>>, @child_chain, @telemetry)
```

Comment:
instead of passing in telemetry, take a look how this can be tested by attaching a handler: https://github.com/mana-ethereum/ethereumex/blob/master/test/ethereumex/counter_test.exs

Reply:
I like that! Updated in https://github.com/omisego/elixir-omg/pull/1470/commits/73941fbc94870efdebb654eda9d99706ee39f870.